cirq/work/pauli_sum_collector.py

```diff
@@ -0,0 +1,103 @@
+# Copyright 2019 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import collections
+from typing import Optional, MutableMapping, cast, Union
+
+import numpy as np
+
+from cirq import circuits, study, ops
+from cirq.work import collector
+
+
+class PauliSumCollector(collector.Collector):
+    """Estimates the energy of a linear combination of Pauli observables."""
+
+    def __init__(self,
+                 circuit: circuits.Circuit,
+                 hamiltonian: ops.PauliSum,
```

Comment:
I don't like the name hamiltonian as that is a the energy observable.  Could we just use pauli_sum or something less specific?

Reply:
Renamed to observable.